e2e/explorer/api/src/app/store.ts

```diff
@@ -155,6 +157,24 @@ function setState(block: Block, receipts: TransactionReceipt[]) {
   return Promise.all(setBalances);
 }
 
+///////////////
+//// BUILD ////
+///////////////
+
+export function setBuildInfo(info: BuildInfo) {
+  const content = {}
+  for (const path of Object.keys(info.input.sources)) {
+    content[path] = {
+      sourceCode: info.input.sources[path].content,
+      contractName: info.input.sources[path],
+      compilerVersion: info.solcVersion,
+      optimizationUsed: info.input.settings.optimizer.enabled,
+      runs: info.input.settings.optimizer.runs
+    }
+  }
+  store.builds = content;
```

Comment:
The problem here, is that you're going to erase the build each time we use `setBuildInfo`. If the application has several buildInfo files, you'll erase the content of the previous one each time you trigger the function.